cmake/SpectreTargetHeaders.cmake

```diff
@@ -0,0 +1,90 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Add header files for CMake to track for a given library
+# or executable.
+#
+# The header and source files together are used to track dependencies between
+# targets automatically in SpECTRE. Specifically, CI checks that the
+# dependencies of a target supply all the header files used in a library.
+# While this checking still requires includes to be correct (be they checked
+# manually or automatically), it significantly lowers the burden of correct
+# dependency management on developers.
+#
+# Additionally, some IDEs perform better if the header files for libraries
+# are known.
+#
+# Usage:
+#  spectre_target_headers(
+#    MyLib
+#    HEADERS
+#    A.hpp
+#    B.hpp
+#    B.tpp
+#    C.hpp)
+#
+# Arguments:
+# - TARGET_NAME: the name of the library or executable
+# - INCLUDE_DIRECTORY: the directory relative to which the header
+#                      files must be included.
```

Comment:
1. In the example, the INCLUDE_DIRECTORY is always `SPECTRE_ROOT/src`. So do I correctly understand this is the relative dir for the `#include "blah"` statements in the C++? The current words are unclear if it is for C++ includes or for the "A.hpp B.hpp" arguments to the CMake funcion as in the usage example above. Can this be clarified?

2. For my understanding, why is this a function argument? Is it to handle libs under other dirs like tests/Helpers and so on? Could this information be easily inferred from the location of the `CMakeLists.txt` file, to avoid the user having to write in this low-information line every time? (I'm asking more for my own education than as a change request)

Reply:
1. Yep, I've attempted to clarify

2. It is needed because header files can specify their own include paths, as you pointed out with `src` and `tests/Unit`. I thought about automating it based on where the header file is, but decided I personally dislike that level of automation because it makes results too magical and difficult to understand :man_shrugging:  (the same reason I dislike defaults in general)